src/index.js

```diff
@@ -0,0 +1,168 @@
+import { differenceCiede2000 } from 'd3-color-difference';
+
+export default class toColor {
+  HUE_MAX = 360;
+
+  constructor(seed, options) {
+    this.options = options || {};
+    if (typeof seed === 'string' || typeof seed === 'number') {
+      this.seed = typeof seed === 'string' ? this._stringToInteger(seed) : seed;
+    } else {
+      throw new TypeError('Seed value must be a number or string');
+    }
+
+    this.known = [];
+  }
+
+  getColor(count = 0) {
+    const h = this._pickHue();
+    const s = this._pickSaturation(h);
+    const b = this._pickBrightness(h, s);
+    const hsl = this._HSVtoHSL(h, s, b);
+    const formatted = this._formatHSL(hsl);
+    const PASSABLE_DISTANCE = 50; 
+
+    // The larger `count` grows, we need to divide actual distance to avoid
+    // hitting a maxiumum call stack error.
+    const ACTUAL_DISTANCE = PASSABLE_DISTANCE / Math.pow(1.25, count);
+
+    // Detect color similarity. If values are too close to one another, call
+    // getColor until enough dissimilarity is acheived.
+    if (this.known.length &&
+        this.known.some(v => differenceCiede2000(v, formatted) < ACTUAL_DISTANCE)) {
+      count++
+      return this.getColor(count);
+    } else {
+      this.known.push(formatted);
+      // Apply modifiers after distribution check + regeneration to ensure 
+      // colors with brightness/saturation adjustments remain the same.
+      return this._colorWithModifiers(h, s, b) 
```

Comment:
Should color modifiers be % based rather than absolute? This would match how, for example, CSS color filters work. I noticed on the demo page, saturation -50 turns many swatches gray.